pkg/webhook/util/filter.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	ctlbackup "github.com/harvester/harvester/pkg/controller/master/backup"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/webhook/indexeres"
+)
+
+func HasInProgressingVMBackupBySourceUID(cache ctlharvesterv1.VirtualMachineBackupCache, sourceUID string) (bool, error) {
+	vmBackups, err := cache.GetByIndex(indexeres.VMBackupBySourceUIDIndex, sourceUID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	for _, vmBackup := range vmBackups {
+		if ctlbackup.IsBackupProgressing(vmBackup) || ctlbackup.GetVMBackupError(vmBackup) != nil {
```

Comment:
Another point is:

Consider using the general used "Status" in VirutalMachine  instance, the "backup" action updates the state of VirtualMachine instance,   the webhook then get the info from VirtualMachine "Status" directly, knows if/not allowing the stop/restart operation.

Reply:
Thanks for the suggestion. Since VMI is a KubeVirt resource, if we want to update it, we may need to update its annotations. In this case, we should maintain data consistency between VMBackup status and VMI annotations. I think it's better to get VMBackup information from its status.